provider/pkg/provider/remote/copyfileController.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"os"
+
+	"github.com/pkg/sftp"
+
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-go-provider/infer"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+)
+
+// These are not required. They indicate to Go that Command implements the following interfaces.
+// If the function signature doesn't match or isn't implemented, we get nice compile time errors in this file.
+var _ = (infer.CustomResource[CopyFileInputs, CopyFileOutputs])((*CopyFile)(nil))
+var _ = (infer.ExplicitDependencies[CopyFileInputs, CopyFileOutputs])((*CopyFile)(nil))
+
+func (r *CopyFile) WireDependencies(f infer.FieldSelector, args *CopyFileInputs, state *CopyFileOutputs) {
+	f.OutputField(&state.CopyFileInputs.Connection).DependsOn(f.InputField(&args.Connection))
+	f.OutputField(&state.CopyFileInputs.Triggers).DependsOn(f.InputField(&args.Triggers))
+	f.OutputField(&state.CopyFileInputs.LocalPath).DependsOn(f.InputField(&args.LocalPath))
+	f.OutputField(&state.CopyFileInputs.RemotePath).DependsOn(f.InputField(&args.RemotePath))
```

Comment:
Do you think there should be an `DependOnMatchingInputs` method on `infer.FieldSelector` to do this without all the boilerplate? The semantics would be `state.A` depends on `args.B` iff the pulumi name of `args.B` matches the pulumi name of `state.A`.

I suspect this will be a common pattern. My only concern is discoverability. @dixler what do you think?

Reply:
IMO I think it's better to force the author to be a little more explicit rather than depend on implicit/magic default behavior. If someone explicitly asks for it in the future, we can add it.

Reply:
Sounds good. BTW, I wasn't suggesting that this becomes the default behavior, I was suggesting that we add a method (that the user needs to call) to do this.

My concern is that if you add a new field to `CopyFileInputs`, you might forget to add it to this block.